api/transport/propagation.go

```diff
@@ -112,10 +118,14 @@ func (e *ExtractOpenTracingSpan) Do(
 
 // UpdateSpanWithErr sets the error tag on a span, if an error is given.
 // Returns the given error
-func UpdateSpanWithErr(span opentracing.Span, err error) error {
+func UpdateSpanWithErr(span opentracing.Span, err error, errCode ...yarpcerrors.Code) error {
```

Comment:
What if we need another argument in the future? I'm afraid it's not a Go-way.

Let's introduce new function like `UpdateSpanWithErrAndCode` that accepts three arguments, and if err != nil, calls UpdateSpanWithErr and span.SetTag.